launcher/src/main/java/org/apache/spark/launcher/InProcessAppHandle.java

```diff
@@ -66,7 +66,7 @@ synchronized void start(String appName, Method main, String[] args) {
         setState(State.FAILED);
       }
 
-      disconnect();
+      dispose();
```

Comment:
If we call `disconnect` here, we would close the connection, and then wait the close to finish in `dispose`. If we call `dispose` directly, we also close and wait the connection(in `waitForClose`). What the actual difference here?

Reply:
The order in which the connection is closed. `waitForClose` will wait for the connection to be closed by the remote side (the finished app) before closing it itself, like disconnect does.

Reply:
ah i see